R/content.R

```diff
@@ -207,16 +207,18 @@ method(format, ContentToolRequest) <- function(
 ) {
   show <- arg_match(show)
 
-  if (length(x@arguments) == 0) {
-    call <- call2(x@name)
-  } else {
-    call <- call2(x@name, !!!x@arguments)
+  arguments <- tool_request_args(x)
+  if (is_tool_result(arguments)) {
+    # Failed to convert the arguments so just use unconverted
+    arguments <- x@arguments
   }
+  call <- call2(x@name, !!!arguments)
+  call_str <- deparse1(call)
```

Comment:
One other edge case we might want to consider here is very long tool call functions. For example, a tool that writes a file will take an input of possibly tens to hundreds of lines of code as a string. As a heuristic, if the call is wrapped onto more than a few lines, we might just want to show the function name or truncate the call with an ellipsis.

Reply:
Yeah, good idea. I hacked in something basic for now.